.github/workflows/validate_urls.yml

```diff
@@ -0,0 +1,39 @@
+name: Validate URLs
+
+on:
+  page_build:
+  schedule:
+    - cron: '00 14,23 * * *'
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Branch to validate'
+        default: 'main'
+        required: true
+
+jobs:
+  validate_documentation_links_on_branch:
+    name: Validate links on branch
+    runs-on: ubuntu-latest
+    env:
+      DEFAULT_BRANCH: main
+    steps:
+    - name: Configure variables
+      run: echo "GIT_BRANCH=${{ github.event.inputs.branch || env.DEFAULT_BRANCH }}" >> $GITHUB_ENV
+    - uses: urlstechie/urlchecker-action@master
```

Comment:
We will need to add this to our accepted actions for SwiftCurrent. I will take that on shortly and comment again when it's done.